### ISSUE TITLE
 Undefined array key 0 in ... loader.php on line 216

### ISSUE THREAD

**Issue by macin** [dokuwiki/dokuwiki#4321]
### The problem

Hi, since some time, I started getting this warnign in my dokuwiki installation when I open configuration page.

```
Warning
: Undefined array key 0 in
C:\<anonymized>\Dokuwiki\DokuWikiStick\dokuwiki\lib\plugins\config\core\Loader.php
on line
216
```
 I cannot trace  back in my memory which plugin I installed started causing this warning . What is the best way to debug this, without having to turn on/off every single plugin (I've got about 30 installed)

### Version of DokuWiki

2024-0206a Kaos

### PHP Version

Dokuwiki Stick version on windows

### Webserver and version of webserver

Dokuwiki Stick version on windows

### Browser and version of browser, operating system running browser

Chrome 128.0.6613.119 (Official Build) (64-bit)

### Additional environment information

this is my local copy of the production installation

**Comment by fiwswe** [comment]
>I cannot trace back in my memory which plugin I installed started causing this warning . What is the best way to debug this, without having to turn on/off every single plugin (I've got about 30 installed)

Do a binary search. Disable half of the enabled plugins. If the problem persists one of the remaining ones is the culprit. Otherwise it was one of those you just disabled. Continue to split the set of enabled plugins in half until you find the one that is causing the issue. Then file a bug for that plugin. With 30 3rd-party plugins installed you would need at most 5 iterations to find the correct one.

This is almost certainly a plugin problem, not a DokuWiki bug. So your issue was filed in the wrong place.

>Webserver and version of webserver
>
>Dokuwiki Stick version on windows

That is not a PHP version. Use e.g. the URL parameter `do=check` to show the PHP version that is being used.

**Comment by splitbrain** [comment]
@macin can you remember which plugin caused this? Would be helpful to try reproducing the issue and then to assess if and how it can be caught or produce a better error message.

**Comment by macin** [comment]
@splitbrain I completely forgot, but still have a local env and did similar exercise - most likely AI Chat is causing that... Interestingly enough, on my production environment I don't get this error.

**Comment by splitbrain** [comment]
Yeah, I am pretty sure that the above fix addresses this issue. Closing.
